File: 微信朋友圈自动点赞.js
/**
 * PBottle RPA demo – please refer to the *process development documentation* for API details.
 * Official website: https://rpa.pbottle.com/
 * Process development documentation: https://rpa.pbottle.com/docs/
 * 
 * Feature description: This script demonstrates how to use RPA to automatically like WeChat Moments posts.
 * It uses image recognition technology to locate UI elements, enabling automated social interactions.
 */

const pbottleRPA = require('./pbottleRPA')     // Import the core PBottle RPA library to access RPA functionality

pbottleRPA.showMsg('The process has started', 'Please turn on your computer sound and pay attention to the running log information.')

pbottleRPA.tts('Preparing to run the batch like script for WeChat Moments. Please log in to WeChat first.')
pbottleRPA.wait(7)

let screenRes = pbottleRPA.getResolution()
pbottleRPA.log('Current screen resolution:', screenRes)

if (screenRes.ratio !== 1) {
    pbottleRPA.tts('Warning: This demo only supports non‑scaled screens.')
    pbottleRPA.showMsg('Warning: This demo only supports non‑scaled screens.')
    pbottleRPA.log('⚠ Warning: This demo only supports non‑scaled screens.')
    pbottleRPA.wait(6)
    pbottleRPA.exit()
}

let color1 = pbottleRPA.getScreenColor(1, screenRes.h - 1);
pbottleRPA.log('System taskbar color:', color1);

pbottleRPA.keyTap('ctrl+alt+w')

// Wait for the target image to appear; timeout is 120 seconds, during which the user is prompted to open the WeChat main window
let position = pbottleRPA.waitImage(['./input/pengYouQuanDianZan/0.png', './input/pengYouQuanDianZan/01.png'], () => {
    pbottleRPA.log('Waiting... Please open the PC version of WeChat first.');
}, 120)

// Open WeChat Moments
pbottleRPA.moveAndClick(position.x, position.y);
pbottleRPA.moveMouse(screenRes.w / 2, screenRes.h / 2);

let counter = 0;
function repeat() {
    pbottleRPA.log('Entering the loop...')
    let result = pbottleRPA.findScreen(['./input/pengYouQuanDianZan/1.png', './input/pengYouQuanDianZan/11.png'], 0.99, 100, 100)

    if (result === false) {
        pbottleRPA.log('Next page')
        pbottleRPA.mouseWheel()
        pbottleRPA.wait(0.5)
        repeat()
    } else {
        pbottleRPA.moveAndClick(result.x, result.y);
        pbottleRPA.wait(0.1)
        
        pbottleRPA.moveAndClick(result.x - 167, result.y);
        counter += 1

        if (counter >= 1000) {
            pbottleRPA.exit('1000 likes are enough, don\'t overdo it! ~')
        }
        pbottleRPA.wait(0.1)
        repeat()
    }
}

repeat()

File: python示例/微信朋友圈自动点赞.py
"""
PBottle RPA Python version (Beta)
https://gitee.com/pbottle/pbottle-rpa
Example
"""

import pbottleRPA  # Import the PBottle RPA module
import time

print("=== WeChat Moments Auto-Like ===")
current_time = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())
print(current_time)

pbottleRPA.showMsg(
    "The process has started",
    "Please turn on your computer sound and pay attention to the running log information.",
)
pbottleRPA.tts(
    "Preparing to run the batch like script for WeChat Moments, adapted for screens without scaling."
)
pbottleRPA.sleep(1000 * 7)


resolution = pbottleRPA.getResolution()
print("Current screen resolution", resolution)
if resolution["ratio"] != 1:
    pbottleRPA.tts("Error: This demo only supports screens with no scaling.")
    print("Error: This demo only supports non‑scaled screens.")
    pbottleRPA.sleep(1000 * 6)
    pbottleRPA.exit()


color = pbottleRPA.getScreenColor(1, resolution["h"] - 1)
print("System taskbar color:", color)


# Note: The image path should not contain custom Chinese characters
rs = pbottleRPA.waitImage(
    ["./input/pengYouQuanDianZan/0.png", "./input/pengYouQuanDianZan/01.png"],
    lambda: print("Waiting... Please open the PC version of WeChat first."),
    120,
)


# Open WeChat Moments
pbottleRPA.moveMouseSmooth(rs["x"], rs["y"])
pbottleRPA.mouseClick()
pbottleRPA.moveMouseSmooth(resolution["w"] / 2, resolution["h"] / 2)

# Like counter
n = 0


# Start the task
def loop():
    print("Entering")
    global n

    rs = pbottleRPA.findScreen(
        ["./input/pengYouQuanDianZan/1.png", "./input/pengYouQuanDianZan/11.png"],
        0.99,
        100,
        100,
    )
    if rs == False:
        print("Next page")
        # pbottleRPA.keyTap('page down')    # WeChat bug – key press often fails to load
        pbottleRPA.mouseWheel()
        pbottleRPA.sleep(500)
        loop()

    else:
        pbottleRPA.moveMouseSmooth(rs["x"], rs["y"])
        pbottleRPA.mouseClick()

        pbottleRPA.sleep(100)

        pbottleRPA.moveMouseSmooth(rs["x"] - 167, rs["y"])
        pbottleRPA.mouseClick()

        # 1000 likes are enough – don’t overdo it
        n += 1
        if n >= 1000:
            pbottleRPA.exit("")

        pbottleRPA.sleep(100)
        loop()  # Repeat


loop()
